src/constructorio-id.js

```diff
@@ -32,6 +32,12 @@
 
         if (this.client_id_storage_location === 'cookie') {
           persisted_id = this.get_cookie(this.cookie_name_client_id);
+
+          if (persisted_id && this.cookie_domain) {
+            // Delete the existing cookie set without a domain and set the same value with a domain
+            this.delete_cookie(this.cookie_name_client_id);
+            this.set_cookie(this.cookie_name_client_id, persisted_id);
```

Comment:
I'm as clear on how this sets a cookie "with a domain", as the domain isn't specified? could you try and explain a bit here (maybe a comment would be ideal in the code for future reference as well).